doc/contents.rst

```diff
@@ -21,6 +21,7 @@ Table of Contents
    getting_help
    ref_material/nest2_vs_3
    ref_material/pynest_apis
+   documentation_workflow
```

Comment:
Wouldn't it be better to put it under 'GETTING INVOLED' after 'Contributing to NEST'?

Reply:
Yes, that would make more sense!